src/FileList/FileExtInfoLoader.cc

```diff
@@ -680,41 +688,76 @@ void FileExtInfoLoader::AddShapeEntries(CARTA::FileInfoExtended& extended_info,
     extended_info.set_depth(depth);
     extended_info.set_stokes(stokes);
 
+    auto* axes_numbers_info = extended_info.mutable_axes_numbers();
+    // Change to 1-based axis indices
+    axes_numbers_info->set_dir_x(direction_axes[0] + 1);
+    axes_numbers_info->set_dir_y(direction_axes[1] + 1);
+    axes_numbers_info->set_spectral(spectral_axis + 1);
+    axes_numbers_info->set_stokes(stokes_axis + 1);
+    axes_numbers_info->set_depth(depth_axis + 1);
+
+    // Set axis types with respect to axis numbers 0~3
+    std::vector<std::string> axis_types(4, "NA");
```

Comment:
I wonder if we could pass in the CTYPE names instead of iterating through all of the header entries.  If not compressed FITS, `_loader->GetCoordinateSystem()->worldAxesNames()` and for compressed FITS you just iterated through the header entries in AddInitialComputedEntries so capture these values.

Reply:
or `image->coordinates()->worldAxesNames()`